sgx-sys/src/ioctl/sgx.rs

```diff
@@ -0,0 +1,85 @@
+// Copyright 2019 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use super::Ioctl;
+use bitflags::bitflags;
+use std::marker::PhantomData;
+use std::os::raw::c_ulong;
+use std::os::unix::io::AsRawFd;
+
+#[repr(C)]
+#[derive(Debug)]
+pub struct Create<'a>(u64, PhantomData<&'a ()>);
+
+impl<'a> Ioctl for Create<'a> {
+    const REQUEST: c_ulong = 1_074_308_096; // SGX_IOC_ENCLAVE_CREATE
+}
+
+impl<'a> Create<'a> {
+    pub fn new(secs: &'a crate::secs::Secs) -> Self {
+        Create(secs as *const _ as _, PhantomData)
```

Comment:
Same for this, I need to learn this syntax.

Reply:
Cast the `Secs` reference to a `Secs` pointer. Then cast the `Secs` pointer to a `u64`. Have Rust infer the types (`_`).